src/main/java/com/fibermc/essentialcommands/commands/FlySpeedCommand.java

```diff
@@ -0,0 +1,77 @@
+package com.fibermc.essentialcommands.commands;
+
+import java.util.Objects;
+
+import com.fibermc.essentialcommands.text.ECText;
+import com.fibermc.essentialcommands.text.TextFormatType;
+
+import com.mojang.brigadier.Command;
+import com.mojang.brigadier.arguments.IntegerArgumentType;
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+
+import net.minecraft.server.command.ServerCommandSource;
+import net.minecraft.server.network.ServerPlayerEntity;
+
+import static com.fibermc.essentialcommands.EssentialCommands.CONFIG;
+
+public class FlySpeedCommand implements Command<ServerCommandSource> {
+    static int speedMultiplier = 20;
+
+    public FlySpeedCommand() {
+    }
+
+    @Override
+    public int run(CommandContext<ServerCommandSource> context) throws CommandSyntaxException {
+        ServerCommandSource source = context.getSource();
+        ServerPlayerEntity targetPlayer = CommandUtil.getCommandTargetPlayer(context);
+
+        int newSpeed = IntegerArgumentType.getInteger(context, "fly_speed");
+
+        exec(source, targetPlayer, newSpeed);
+        return 0;
+    }
+
+    public int reset(CommandContext<ServerCommandSource> context) throws CommandSyntaxException {
+        ServerCommandSource source = context.getSource();
+        ServerPlayerEntity player = source.getPlayer();
```

Comment:
This will mean that the target player for `reset` is always the sender, which I think will be problematic because of:

```java

            flySpeedBuilder
                .requires(permissionSelf)
                .executes(new FlySpeedCommand()::reset)
                .then(CommandManager.literal("reset")
                    .executes(new FlySpeedCommand()::reset))
                .then(argument("fly_speed", IntegerArgumentType.integer(0))
                    .executes(new FlySpeedCommand()))
                .then(CommandUtil.targetPlayerArgument()
                    .requires(permissionOther)
                    .then(CommandManager.literal("reset")
                        .executes(new FlySpeedCommand()::reset)) // <-- this thing
                    .then(argument("fly_speed", IntegerArgumentType.integer(0))
                        .executes(new FlySpeedCommand())));
```

Reply:
Nice catch, hadn't really looked at this(the rest part) after restoring the stash from half a year ago